File: LC-1290/LC-1290-2.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */

/**
 * @param {ListNode} head
 * @return {number}
 */
var getDecimalValue = function (head) {
  /**
   * Concepts: Iterative
   */

  if (head.next === null) return head.val

  let size = 0,
    temp = head
  while (temp !== null) {
    ++size
    temp = temp.next
  }

  --size
  temp = head

  let value = 0
  while (temp != null) {
    if (temp.val) {
      value += Math.pow(2, size)
    }
    --size
    temp = temp.next
  }
  return value
}

File: LC-1290/LC-1290-2.cpp
using namespace std;

/**
 * Definition for singly-linked list.
 */
struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution
{
public:
    /**
     * Concepts: Bitwise Approach
     * Formula: num = num << 1 | x
     */
    int getDecimalValue(ListNode *head)
    {
        int val = 0;
        ListNode *curr = head;
        while (curr != nullptr)
        {
            val = val << 1 | curr->val;
            curr = curr->next;
        }
        return val;
    }
};